prov/verbs/src/verbs_info.c

```diff
@@ -1349,40 +1349,20 @@ static int vrb_device_has_ipoib_addr(const char *dev_name)
 
 #define VERBS_NUM_DOMAIN_TYPES		3
 
-static int vrb_init_info(const struct fi_info **all_infos)
+static int vrb_init_info(struct dlist_entry *verbs_devs,
+			 const struct fi_info **all_infos)
 {
 	struct ibv_context **ctx_list;
 	struct fi_info *fi = NULL, *tail = NULL;
 	const struct verbs_ep_domain *ep_type[VERBS_NUM_DOMAIN_TYPES];
 	int ret = 0, i, j, num_devices, dom_count = 0;
-	static bool initialized = false;
 
 	vrb_prof_func_start(__func__);
 
-	if (initialized)
-		goto done;
-
-	initialized = true;
+	vrb_devs_free(verbs_devs);
+	fi_freeinfo((void *)*all_infos);
```

Comment:
Cast shouldn't be needed.  Please update the function signature to remove the const.

Reply:
If we want to remove the "const" in this function, we will also have to change the definition of `util_prov::info` and remove the "const" from there. Here is how the structure is defined right now:
```
struct util_prov {
       const struct fi_provider        *prov;
       const struct fi_info            *info;    <--- here.
       ofi_alter_info_t                alter_defaults;
       const int                       flags;
};
```

Is it what you want?

Reply:
Hmm... when that was defined, I think the assumption was that the info list was a static list of structures.  But verbs needs to set this dynamically, including freeing it.  Maybe that const should be removed.  Can you tell if removing it causes an issue in a bunch of other places?

Reply:
Removing the "const" only causes an issue to the verbs provider. I added a commit to remove it as a consequence.